python/semantic_kernel/core_plugins/sessions_python_tool/README.md

```diff
@@ -0,0 +1,132 @@
+
+## Authentication to ARM (management.azure.com)
+
+For any call to ARM (management.azure.com), use the access token retrieved from the below call:
+
+```az account get-access-token --resource <https://management.azure.com/>```
+
+## Generate a Session Pool
+
+a. Call the following API to generate a Session Pool:
+
+```PUT <https://management.azure.com/subscriptions/{{SubscriptionId}}/resourceGroups/{{ResourceGroup}}/providers/Microsoft.App/sessionPools/{{SessionPoolName}}?api-version=2023-08-01-preview>```
+
+Body properties:
+
+- location: Azure Region
+- properties:
+  - poolManagementType:
+    - Today there are two Pool Management Types supported:
+           - "Manual"
+                - In this model, the user will call generateSessions API which supports batch mode (to generate 100s of sessions in one API call, and then user is free to update/specialize the session as needed or execute code in the session)
+           - "Dynamic"
+                - In this mode, the pool management is handled by the platform. Currently, the dynamic mode is only implemented for Python code execution scenario, which has its own APIs to execute code.
+  - maxConcurrentSessions:
+    - Maximum number of active sessions allowed
+  - name:
+    - Name of the sessions pool
+  - dynamicPoolConfiguration: Specifies the type of sessions generated by the platform
+    - poolType: Type of images used for the pool
+      - Valid values ["JupyterPython", "FunctionsPython"]
+    - executionType:
+      - Valid values ["Timed"]
+    - coolDownPeriodSeconds:
+      - Integer representing the maximum time allowed before the platform scales down the container
+  - sessionPoolSecrets: Secrets associated with the Session Pool
+    - name: Name of the secret
+    - value: Secret Value
+
+Example Generation of Session Pool:
+
+```json
+{
+    "location": "koreacentral",
+    "properties": {
+        "poolManagementType": "Dynamic",
+        "maxConcurrentSessions": 10,
+        "name": "{{SessionPoolName}}",
+        "dynamicPoolConfiguration": {
+            "poolType": "JupyterPython",
+            "executionType": "Timed",
+            "coolDownPeriodInSecond": 310
+        }
+    }
+}
+```
+
+Curl Example:
+
+```curl
+curl -X PUT "https://management.azure.com/subscriptions/{{SubscriptionId}}/resourceGroups/{{ResourceGroup}}/providers/Microsoft.App/sessionPools/{{SessionPoolName}}?api-version=2023-08-01-preview" \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $token" \
+    -d '{"location": "koreacentral","properties": { "poolManagementType": "Dynamic", "maxConcurrentSessions": 10, "name": "{{SessionPoolName}}", "dynamicPoolConfiguration": { "poolType": "JupyterPython", "executionType": "Timed", "coolDownPeriodInSecond": 310 } } }'
+```
+
+If all goes well, you should receive a 200 Status Code. The response will contain a `poolManagementEndpoint` which is required to configure the Python Plugin below.
```

Comment:
I'm wondering if it would be better to link to the Container Apps docs on creating a session pool rather than have this here. My concern is that it could get stale quickly and we won't know it. @anthonychu do have an opinion on this?

Reply:
Yeah, it's a good callout. Per our offline chat, I will update the README once ACA has public docs we can link to.